src/js/model_features.js

```diff
@@ -0,0 +1,69 @@
+import lodash from "lodash";
+
+import modelFeatures from "../../model_feature_data";
+import { recursiveMerge } from "./utils";
+
+/**
+ * Filter nodes from the model tree based on path and modify node data.
+ * A node of the model tree is defined in mode.js and has a form such as the following:
+ * ```
+ * {
+ *     path: "/dft/gga",
+ *     parentPath: "/dft",
+ *     data: {
+ *         subtype: {
+ *             slug: "gga",
+ *             name: "Generalized Gradient Approximation",
+ *         },
+ *     },
+ *     children: [
+ *         // array of child nodes
+ *     ],
+ * }
+ * ```
+ * @param {Array} nodes - Array of nodes to be filtered.
+ * @param {string[]} paths - Array of node paths
+ * @param {Object[]} pathData - Array of objects containing path and node data.
+ * @returns {Object[]} - The filtered tree (with possibly modified data).
+ */
+export function filterTree(nodes, paths, pathData = null) {
```

Comment:
Now that there is an explanation about what node is, I think this function should go to `mode`

Reply:
Contextually it could be located both in `mode` and in `application-flavors`, as the function acts on the model tree, but requires input defined in `application-flavors`. However, moving this function to `mode` will create a dependency of `application-flavors` to `mode` (right now there is none). Should I still move it?